salt/utils/cloud.py

```diff
@@ -1366,7 +1366,13 @@ def deploy_windows(host,
             if ret_code != 0:
                 return False
 
-            time.sleep(5)
+            for retry in range(100):
+                stdout, stderr, ret_code = run_psexec_command(
+                    'cmd.exe', '/c sc query salt-minion', host, username, password
+                )
+                log.debug('Cloud winrm command #{}: {}'.format(retry, stdout))
+                if 'STOP_PENDING' not in stdout:
+                    break
```

Comment:
Would it make sense to put a momentary or increasing delay in here? I don't know that much about the psexec thing, but it's not obvious if this will just burn through the calls as fast as it can.